weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/ManageColumnsButton.tsx

```diff
@@ -0,0 +1,192 @@
+/**
+ * Button/popup for grid column visibility state.
+ */
+
+import {Popover} from '@mui/material';
+import {GridColDef, GridColumnVisibilityModel} from '@mui/x-data-grid-pro';
+import {Switch} from '@wandb/weave/components';
+import classNames from 'classnames';
+import React, {useCallback, useRef, useState} from 'react';
+
+import {Button} from '../../../../../Button';
+import {DraggableGrow} from '../../../../../DraggablePopups';
+import {TextField} from '../../../../../Form/TextField';
+import {Tailwind} from '../../../../../Tailwind';
+import {ColumnInfo} from '../../types';
+import {TraceCallSchema} from '../wfReactInterface/traceServerClient';
+
+type ManageColumnsButtonProps = {
+  columnInfo: ColumnInfo;
+  columnVisibilityModel: GridColumnVisibilityModel;
+  setColumnVisibilityModel: (model: GridColumnVisibilityModel) => void;
+};
+
+export const ManageColumnsButton = ({
+  columnInfo,
+  columnVisibilityModel,
+  setColumnVisibilityModel,
+}: ManageColumnsButtonProps) => {
+  const [search, setSearch] = useState('');
+  const ref = useRef<HTMLDivElement>(null);
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const onClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(anchorEl ? null : ref.current);
+    setSearch('');
+  };
+
+  const open = Boolean(anchorEl);
+  const id = open ? 'simple-popper' : undefined;
+
+  const toggleColumnShow = useCallback(
+    (key: string) => {
+      const on = columnVisibilityModel[key] ?? true;
+      const newModel = {
+        ...columnVisibilityModel,
+        [key]: !on,
+      };
+      setColumnVisibilityModel(newModel);
+    },
+    [columnVisibilityModel, setColumnVisibilityModel]
+  );
+  const onHideAll = () => {
```

Comment:
when i click this i get a strange blank space in my data grid... 
<img width="732" alt="Screenshot 2024-07-09 at 9 02 53 PM" src="https://github.com/wandb/weave/assets/19414170/209a6c3f-1bcf-4b61-9c21-a2fc0746b2b6">

Reply:
Yes, you can get this in prod too with the Material UI column selector, so unrelated to this PR. I think it is a grid bug in the interaction between hidden and pinned columns (the blank space corresponds to the pinned but hidden feedback column).